### PR TITLE
Topic/unit test info

### PR DESCRIPTION
**Pull request by muellmusik**
Purpose and Motivation
----------------------

Add info on test expectations and links to info on unit test best practice

Types of changes
----------------

- Documentation (non-code change which corrects or adds documentation for existing features)

Checklist
---------

- [x] All tests are passing
- [x] If necessary, new tests were created to address changes in PR, and tests are passing
- [x] Updated documentation, if necessary
- [x] This PR is ready for review

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -24,12 +24,12 @@ Checklist
...
 
-- [ ] All tests are passing
-- [ ] If necessary, new tests were created to address changes in PR, and tests are passing
+- [ ] All previous tests are passing
+- [ ] Tests were created to address all code changes in this PR (including bug fixes), and tests are passing
 - [ ] Updated documentation, if necessary
 - [ ] This PR is ready for review
 
-<!--- See DEVELOPING.md for instructions on running and writing tests. -->
+<!--- See DEVELOPING.md and https://github.com/supercollider/supercollider/wiki/Unit-Testing-Guide for instructions on running and writing tests. -->
```
Comment (snappizz) [inline .github/PULL_REQUEST_TEMPLATE.md]: hmm, must disagree with putting a URL here. this template is already too long, and you have to copy-paste the link.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -24,12 +24,12 @@ Checklist
 
 <!--- Complete an item by checking it: [x] -->
 
-- [ ] All tests are passing
-- [ ] If necessary, new tests were created to address changes in PR, and tests are passing
+- [ ] All previous tests are passing
+- [ ] Tests were created to address all code changes in this PR (including bug fixes), and tests are passing
```
Comment (snappizz) [inline .github/PULL_REQUEST_TEMPLATE.md]: let's keep this short. i suggest:

- [ ] All tests are passing
- [ ] All code changes meet testing guidelines (if applicable)

Comment (patrickdupuis) [reply .github/PULL_REQUEST_TEMPLATE.md]: I like the above, though I would write `(incl. bug fixes)`.

Comment (brianlheim) [reply .github/PULL_REQUEST_TEMPLATE.md]: "incl" is an uncommon abbreviation and is likely to be confusing for people whose first language is not English, so I'd avoid it. Also, not all changes to code will require _new_ tests to be written, thus the original phrasing, "If necessary." My suggestion is "Tests were updated or created to address changes in this PR."

Comment (muellmusik) [reply .github/PULL_REQUEST_TEMPLATE.md]: Okay, sounds reasonable.

**Comment by muellmusik** [comment]
I’m not sure why Travis should have failed...

**Comment by muellmusik** [comment]
Looks like unit tests

**Comment by brianlheim** [comment]
@muellmusik you have the ability to restart CI builds - login to Travis and AppVeyor with your github credentials (choose "SuperCollider" account in AppVeyor)

**Comment by muellmusik** [comment]
@brianlheim done in travis; can't see where to restart in AppVeyor. I was more wondering why a commit to an md file should cause builds to fail

**Comment by brianlheim** [comment]
Simple version of the answer - because our CI script sometimes fails for no good reason. Something to do with buffering.

Re AppVeyor if you follow the link from the failing build here and are signed in you will see a "rebuild commit" button.

**Comment by muellmusik** [comment]
> Re AppVeyor if you follow the link from the failing build here and are signed into the supercollider acct you will see a "rebuild commit" button. Easy way to check kf you're signed into the right acct is by going to Projects and you should see sc and sc3 plugins as the projects available.

I can see those, but can't see the rebuild button. Anyway, it's irrelevant for this PR, oder?

**Comment by muellmusik** [comment]
This is still failing on the Qt5.9 App Store build, but none of the changes touches compilable code, so I think it’s irrelevant. Is it possible to finish review and merge?

**Comment by brianlheim** [comment]
Yes, it's fine, the failure is just due to an old version of the appveyor script on your feature branch. Sorry that that noise was all over this PR. Also wanted to note that Core Team group members are definitely now able to restart AppVeyor builds; I had missed a checkbox within Appveyor's web UI. :) Thanks for your patience everyone!

**Pull request merged**